### ISSUE TITLE
Joplin crashes when trying to Change application layout

### ISSUE THREAD

**Issue by marmonto** [laurent22/joplin#5111]
After updating from 1.8.5 to 2.0.11, I wanted to reorder my layout.
When I try to change application layout, application crashes when it affects the main editor section. I can move any other component freely but that action can't affect main editor.

Steps:
1. In ` View ` menu, select ` Change application layout ` option.
2. Try to relocate any component in the layout.
3. Joplin continuous working without crashing.
3.1. If I try to relocate the ` Main editor ` section, Joplin crashes.
3.2. If I try to relocate any section ( Menu, Favorites ), Joplin works, but if the relocation affects ` Main editor `, Joplin crashes. 

I though that it could be caused by a Plugin, but no. I checked after disabling and finally uninstalling them all. Then I have only reinstalled ` Favorites ` plugin.

## Environment

Joplin version: Joplin Desktop
Platform: Windows
OS specifics: 10 (21H1 - 19043.1055)

## Steps to reproduce

1. Go to menu and click View, then select sub-option _Change application layout_
2. Try to reorder items. All works until central panel (editor) is moved, Joplin crash. 

## Describe what you expected to happen

Reorder layout objects, as usual.
After updating from 1.8.5 to 2.0.11, I wanted to reorder my layout.
I have uninstalled some plugins also and it persits.
I haven't personalized it using the _css_ files.

## Logfile

When the error occurs, a page window appears doing impossible to continue working. That's the attached log. I also started Joplin by enabling debugging flags using the ` flags.txt ` file, but I found same messages in the console.
I don't have experience in Joplin development, and Electron either, but I saw that it is continuously looking for a _resources_ folder in install folder ` D:\Program Files (Medea)\Joplin\resources\app.asar\ [FILES] `, but ` app.asar ` is a file and not a folder (I think everything is inside, like a library, don't know hehe).

## Environment

Joplin version: Joplin Desktop
Platform: Windows
OS specifics: 10 (21H1 - 19043.1055)

## Steps to reproduce

1. Go to menu and click View, then select sub-option _Change application layout_
2. Try to reorder items. All works until central panel (editor) is moved, Joplin crash. 

## Describe what you expected to happen

Reorder layout objects, as usual.
After updating from 1.8.5 to 2.0.11, I wanted to reorder my layout.
I have uninstalled some plugins also and it persits.
I haven't personalized it using the _css_ files.

### Error

Joplin encountered a fatal error and could not continue. To report the error, please copy the *entire content* of this page and post it on Joplin forum or GitHub.

To continue you may close the app. Alternatively, if the error persists you may try to restart in safe mode, which will temporarily disable all plugins.

### Message

Cannot read property 'maxSize' of null

### Version info

Joplin for Desktop

Copyright © 2016-2021 Laurent Cozic
Joplin 2.0.11 (prod, win32)

Client ID: 14362837e7d44f8dad0ca81379d4d6bc
Sync Version: 2
Profile Version: 39
Keychain Supported: Yes

Revision: 981b1d845

### Plugins

[
    {
        "id": "io.github.jackgruber.combine-notes",
        "name": "Combine notes",
        "enabled": true,
        "version": "0.2.2"
    },
    {
        "id": "joplin.plugin.ambrt.convertToNewNote",
        "name": "Convert Text To New Note",
        "enabled": true,
        "version": "1.5.0"
    },
    {
        "id": "joplin.plugin.ambrt.goToItem",
        "name": "Create and go to #tags and @notebooks",
        "enabled": true,
        "version": "1.3.7"
    },
    {
        "id": "joplin.plugin.ambrt.embedSearch",
        "name": "Embed Search",
        "enabled": true,
        "version": "1.0.11"
    },
    {
        "id": "joplin.plugin.note.tabs",
        "name": "Note Tabs",
        "enabled": false,
        "version": "1.3.0"
    },
    {
        "id": "joplin.plugin.benji.favorites",
        "name": "Favorites",
        "enabled": true,
        "version": "1.2.1"
    },
    {
        "id": "com.export-to-ssg.aman-d-1-n-only",
        "name": "Export To SSG",
        "enabled": true,
        "version": "1.0.3"
    },
    {
        "id": "joplin.plugin.ambrt.fold-cm",
        "name": "Folding in Code Mirror Editor",
        "enabled": true,
        "version": "1.2.5"
    },
    {
        "id": "joplin.plugin.anki-sync",
        "name": "Joplin Anki Sync",
        "enabled": true,
        "version": "1.0.0"
    },
    {
        "id": "com.whatever.inline-tags",
        "name": "Inline tags",
        "enabled": true,
        "version": "1.1.3"
    },
    {
        "id": "plugin.azamahJunior.note-statistics",
        "name": "Note Statistics",
        "enabled": true,
        "version": "1.0.4"
    },
    {
        "id": "io.github.jackgruber.note-overview",
        "name": "Note overview",
        "enabled": true,
        "version": "1.4.3"
    },
    {
        "id": "joplin.plugin.spoiler.cards",
        "name": "Spoilers",
        "enabled": true,
        "version": "1.0.3"
    },
    {
        "id": "plugin.calebjohn.rich-markdown",
        "name": "Rich Markdown",
        "enabled": true,
        "version": "0.4.7"
    }
]

### Stack trace

TypeError: Cannot read property 'maxSize' of null
    at renderLayoutItem (D:\Program Files (Medea)\Joplin\resources\app.asar\gui\ResizableLayout\ResizableLayout.js:86:139)
    at renderLayoutItem (D:\Program Files (Medea)\Joplin\resources\app.asar\gui\ResizableLayout\ResizableLayout.js:102:41)
    at ResizableLayout (D:\Program Files (Medea)\Joplin\resources\app.asar\gui\ResizableLayout\ResizableLayout.js:117:22)
    at renderWithHooks (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:15108:18)
    at updateFunctionComponent (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:16925:20)
    at beginWork$1 (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:18498:16)
    at HTMLUnknownElement.callCallback (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:347:14)
    at Object.invokeGuardedCallbackDev (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:397:16)
    at invokeGuardedCallback (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:454:31)
    at beginWork$$1 (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:23217:7)

### Component stack

    in ResizableLayout (created by MainScreenComponent)
    in div (created by MainScreenComponent)
    in MainScreenComponent (created by Connect(MainScreenComponent))
    in Connect(MainScreenComponent) (created by NavigatorComponent)
    in div (created by NavigatorComponent)
    in NavigatorComponent (created by Connect(NavigatorComponent))
    in Connect(NavigatorComponent) (created by RootComponent)
    in ThemeProvider (created by RootComponent)
    in StyleSheetManager (created by RootComponent)
    in RootComponent (created by Connect(RootComponent))
    in Connect(RootComponent)
    in ErrorBoundary
    in Provider
	
React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary.

**Comment by laurent22** [comment]
@mablin7, the error seems to indicate an issue with the recent layout change. Any idea what might be the issue?

```
TypeError: Cannot read property 'maxSize' of null
at renderLayoutItem (D:\Program Files (Medea)\Joplin\resources\app.asar\gui\ResizableLayout\ResizableLayout.js:86:139)
at renderLayoutItem (D:\Program Files (Medea)\Joplin\resources\app.asar\gui\ResizableLayout\ResizableLayout.js:102:41)
at ResizableLayout (D:\Program Files (Medea)\Joplin\resources\app.asar\gui\ResizableLayout\ResizableLayout.js:117:22)
at renderWithHooks (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:15108:18)
at updateFunctionComponent (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:16925:20)
at beginWork$1 (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:18498:16)
at HTMLUnknownElement.callCallback (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:347:14)
at Object.invokeGuardedCallbackDev (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:397:16)
at invokeGuardedCallback (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:454:31)
at beginWork$$1 (D:\Program Files (Medea)\Joplin\resources\app.asar\node_modules\react-dom\cjs\react-dom.development.js:23217:7)

```

**Comment by mablin7** [comment]
Yes, it's most likely my bad. I'll look into it today.

**Comment by mablin7** [comment]
Hm I cannot reproduce this on either 2.0.11 or the dev brach. Could you attach here your settings.json file? I suspect this has to do with the plugins you had, even if you uninstalled them. As far as I know, their panels remain in the layout.

Here's how to find the settings.json file:

1. Open joplin
2. Click Help > Open profile directory
3. Copy the contents of `settings.json` here, or upload the file.

I don't think it can contain any personal information, but you should take a look before sending. I'm only interested in the `ui.layout` section

**Comment by marmonto** [comment]
> 
> 
> Hm I cannot reproduce this on either 2.0.11 or the dev brach. Could you attach here your settings.json file? I suspect this has to do with the plugins you had, even if you uninstalled them. As far as I know, their panels remain in the layout.
> 
> Here's how to find the settings.json file:
> 
>     1. Open joplin
> 
>     2. Click Help > Open profile directory
> 
>     3. Copy the contents of `settings.json` here, or upload the file.
> 
> 
> I don't think it can contain any personal information, but you should take a look before sending. I'm only interested in the `ui.layout` section


Here it is, my ` settings.json ` file

[settings.json.txt](https://github.com/laurent22/joplin/files/6695312/settings.json.txt)

**Comment by marmonto** [comment]
Hi.

I renamed the `settings.json` file and I started Joplin; as I imagined, the file was recreated and, at least, all my data stills there and organized. Then, I moved to change the layout and everything worked **perfect**. I took the risk to compare previous and new file and I still without understand it (well, I know, I don't have any idea about what Joplin is looking for).

Well, I expect this gives you any clue about what could happened and I'll move to continue playing and enjoying _Joplin_ .

Note: Do I have to close the bug? Please, let me know ....or feel free to do it.

**Comment by mablin7** [comment]
Hey sorry I haven't had time to really look into this, but I surely will. You shouldn't close this issue as the problem still exists, but I'm glad you found a workaround.

**Comment by marmonto** [comment]
> 
> 
> Hey sorry I haven't had time to really look into this, but I surely will. You shouldn't close this issue as the problem still exists, but I'm glad you found a workaround.

Ok @mablin7, thank you.
I'll be paying attention in case there is something else I can help.

**Comment by laurent22** [comment]
@mablin7, if the bug cannot be replicated, is there any way to add a null check on the line where the error happened? It seems null items sometimes make their way into the layout but it's not clear how. However if we can at least prevent it from crashing that would be a start.

**Comment by mablin7** [comment]
Right, that sounds reasonable, I'll do it

**Comment by amandamcg** [comment]
I hit this issue also, I am on Mac Joplin 2.1.8 (prod, darwin). 

tl;dr: I was able to fix this by deleting the `ui.layout` contents in my settings.json file. I am including the rest as repro steps in case it helps.


---
## Repro steps

(Note: I recently had an [SQL issue](https://github.com/laurent22/joplin/issues/5153) and so it is possible that my app just got into a strange state while restoring, so please take this report with a grain of salt)
- I have lots of plugins installed, and I recently disabled all of them (by renaming plugin folder) and reenabled them (by changing the name back). The plugins no longer were in my old layout, each was in its own column. When I tried to move them around, I got a crash.
<details>
<summary>Crash log - Cannot read property 'maxSize' of null (plugins installed)</summary>

```
Error
Joplin encountered a fatal error and could not continue. To report the error, please copy the *entire content* of this page and post it on Joplin forum or GitHub.

To continue you may close the app. Alternatively, if the error persists you may try to restart in safe mode, which will temporarily disable all plugins.

Message
Cannot read property 'maxSize' of null

Version info
Joplin for Desktop

Copyright © 2016-2021 Laurent Cozic
Joplin 2.1.8 (prod, darwin)

Client ID: xxx
Sync Version: 2
Profile Version: 39
Keychain Supported: Yes

Revision: 94ea177
Plugins
[
    {
        "id": "joplin.plugin.ambrt.backlinksToNote",
        "name": "Automatic Backlinks to note",
        "enabled": true,
        "version": "2.0.10"
    },
    {
        "id": "joplin.plugin.note.tabs",
        "name": "Note Tabs",
        "enabled": true,
        "version": "1.3.0"
    },
    {
        "id": "outline",
        "name": "Outline",
        "enabled": true,
        "version": "1.2.1"
    },
    {
        "id": "joplin.plugin.ambrt.goToItem",
        "name": "Create and go to #tags and @notebooks",
        "enabled": false,
        "version": "1.3.7"
    },
    {
        "id": "joplin.plugin.ambrt.convertToNewNote",
        "name": "Convert Text To New Note",
        "enabled": true,
        "version": "1.5.0"
    },
    {
        "id": "io.github.jackgruber.combine-notes",
        "name": "Combine notes",
        "enabled": true,
        "version": "0.2.2"
    },
    {
        "id": "io.github.jackgruber.note-overview",
        "name": "Note overview",
        "enabled": true,
        "version": "1.4.3"
    },
    {
        "id": "plugin.calebjohn.MathMode",
        "name": "Math Mode",
        "enabled": true,
        "version": "0.5.0"
    },
    {
        "id": "io.github.jackgruber.copytags",
        "name": "Tagging",
        "enabled": true,
        "version": "1.0.1"
    },
    {
        "id": "com.whatever.inline-tags",
        "name": "Inline tags",
        "enabled": true,
        "version": "1.1.3"
    },
    {
        "id": "com.whatever.quick-links",
        "name": "Quick Links",
        "enabled": true,
        "version": "1.2.1"
    },
    {
        "id": "joplin.plugin.benji.favorites",
        "name": "Favorites",
        "enabled": true,
        "version": "1.2.1"
    },
    {
        "id": "joplin.plugin.forcewake.tags-generator",
        "name": "Tags Generator",
        "enabled": true,
        "version": "1.0.0"
    },
    {
        "id": "joplin.plugin.ambrt.copyNoteLink",
        "name": "Copy link to active note",
        "enabled": true,
        "version": "1.0.1"
    },
    {
        "id": "io.treymo.LinkGraph",
        "name": "Link Graph UI",
        "enabled": false,
        "version": "0.9.9"
    },
    {
        "id": "plugin.calebjohn.rich-markdown",
        "name": "Rich Markdown",
        "enabled": true,
        "version": "0.4.5"
    },
    {
        "id": "joplin.plugin.benji.quick-move",
        "name": "Quick Move",
        "enabled": false,
        "version": "1.0.0"
    },
    {
        "id": "joplin.plugin.ambrt.fold-cm",
        "name": "Folding in Code Mirror Editor",
        "enabled": false,
        "version": "1.2.5"
    },
    {
        "id": "plugin.azamahJunior.note-statistics",
        "name": "Note Statistics",
        "enabled": true,
        "version": "1.0.4"
    },
    {
        "id": "com.plugin.randomNotePlugin",
        "name": "Random note",
        "enabled": true,
        "version": "1.0.0"
    },
    {
        "id": "net.rmusin.joplin-table-formatter",
        "name": "Table Formatter Plugin",
        "enabled": true,
        "version": "1.1.2"
    },
    {
        "id": "com.github.uphy.PlantUmlPlugin",
        "name": "PlantUML",
        "enabled": true,
        "version": "0.0.1"
    },
    {
        "id": "joplin.plugin.benji.persistentLayout",
        "name": "Persistent Editor Layout",
        "enabled": true,
        "version": "2.0.0"
    }
]
Stack trace
TypeError: Cannot read property 'maxSize' of null
    at renderLayoutItem (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ResizableLayout/ResizableLayout.js:86:139)
    at renderLayoutItem (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ResizableLayout/ResizableLayout.js:102:41)
    at ResizableLayout (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ResizableLayout/ResizableLayout.js:117:22)
    at renderWithHooks (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:15108:18)
    at updateFunctionComponent (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:16925:20)
    at beginWork$1 (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:18498:16)
    at HTMLUnknownElement.callCallback (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:347:14)
    at Object.invokeGuardedCallbackDev (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:397:16)
    at invokeGuardedCallback (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:454:31)
    at beginWork$$1 (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:23217:7)
Component stack

    in ResizableLayout (created by MainScreenComponent)
    in div (created by MainScreenComponent)
    in MainScreenComponent (created by Connect(MainScreenComponent))
    in Connect(MainScreenComponent) (created by NavigatorComponent)
    in div (created by NavigatorComponent)
    in NavigatorComponent (created by Connect(NavigatorComponent))
    in Connect(NavigatorComponent) (created by RootComponent)
    in ThemeProvider (created by RootComponent)
    in StyleSheetManager (created by RootComponent)
    in RootComponent (created by Connect(RootComponent))
    in Connect(RootComponent)
    in ErrorBoundary
    in Provider
```
</details>

---

- Starting the app in safe mode did not help. Even without the plugins loaded, the app still crashed when rearranging the columns.
<details>
<summary>Crash log - Cannot read property 'maxSize' of null (jn safe mode)</summary>

```
Error
Joplin encountered a fatal error and could not continue. To report the error, please copy the *entire content* of this page and post it on Joplin forum or GitHub.

To continue you may close the app. Alternatively, if the error persists you may try to restart in safe mode, which will temporarily disable all plugins.

Message
Cannot read property 'maxSize' of null

Version info
Joplin for Desktop

Copyright © 2016-2021 Laurent Cozic
Joplin 2.1.8 (prod, darwin)

Client ID: xxx
Sync Version: 2
Profile Version: 39
Keychain Supported: Yes

Revision: 94ea177
Stack trace
TypeError: Cannot read property 'maxSize' of null
    at renderLayoutItem (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ResizableLayout/ResizableLayout.js:86:139)
    at renderLayoutItem (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ResizableLayout/ResizableLayout.js:102:41)
    at ResizableLayout (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ResizableLayout/ResizableLayout.js:117:22)
    at renderWithHooks (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:15108:18)
    at updateFunctionComponent (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:16925:20)
    at beginWork$1 (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:18498:16)
    at HTMLUnknownElement.callCallback (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:347:14)
    at Object.invokeGuardedCallbackDev (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:397:16)
    at invokeGuardedCallback (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:454:31)
    at beginWork$$1 (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:23217:7)
Component stack

    in ResizableLayout (created by MainScreenComponent)
    in div (created by MainScreenComponent)
    in MainScreenComponent (created by Connect(MainScreenComponent))
    in Connect(MainScreenComponent) (created by NavigatorComponent)
    in div (created by NavigatorComponent)
    in NavigatorComponent (created by Connect(NavigatorComponent))
    in Connect(NavigatorComponent) (created by RootComponent)
    in ThemeProvider (created by RootComponent)
    in StyleSheetManager (created by RootComponent)
    in RootComponent (created by Connect(RootComponent))
    in Connect(RootComponent)
    in ErrorBoundary
    in Provider


```
</details>

---

- Even after completely deleting the plugin folder and being in safe mode,  I still see the following error in the dev tools logs
<details>
<summary>Crash log - Could not get plugin info: Error: Plugin not found: joplin.plugin.ambrt.backlinksToNote </summary>

```
/Applications/Joplin.app/Contents/Resources/app.asar/gui/ErrorBoundary.js:40 Could not get plugin info: Error: Plugin not found: joplin.plugin.ambrt.backlinksToNote
    at PluginService.pluginById (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/services/plugins/PluginService.js:84)
    at ErrorBoundary.componentDidCatch (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ErrorBoundary.js:30)
    at ErrorBoundary.callback (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:20965)
    at callCallback (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:12929)
    at commitUpdateEffects (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:12968)
    at commitUpdateQueue (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:12959)
    at commitLifeCycles (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:20079)
    at commitLayoutEffects (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:22813)
    at HTMLUnknownElement.callCallback (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:347)
    at Object.invokeGuardedCallbackDev (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:397)
componentDidCatch @ /Applications/Joplin.app/Contents/Resources/app.asar/gui/ErrorBoundary.js:40
callback @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:20965
callCallback @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:12929
commitUpdateEffects @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:12968
commitUpdateQueue @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:12959
commitLifeCycles @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:20079
commitLayoutEffects @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:22813
callCallback @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:347
invokeGuardedCallbackDev @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:397
invokeGuardedCallback @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:454
commitRootImpl @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:22585
unstable_runWithPriority @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/scheduler/cjs/scheduler.development.js:643
runWithPriority$2 @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11305
commitRoot @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:22414
runRootCallback @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:21554
(anonymous) @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11353
unstable_runWithPriority @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/scheduler/cjs/scheduler.development.js:643
runWithPriority$2 @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11305
flushSyncCallbackQueueImpl @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11349
flushSyncCallbackQueue @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11338
discreteUpdates$1 @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:21677
discreteUpdates @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:2359
dispatchDiscreteEvent @ /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:5979
```
</details>

---

- I reread the notes from the rest of the thread and looked at the ` ui.layout` part of settings. NOTE: Deleting the contents of this fixes the crash

<details>
<summary>Settings ui.layout </summary>
```
"ui.layout": {
		"key": "root",
		"children": [
			{
				"key": "tempContainer-5WW9QHxTwhJDWLEtE3IEzZ",
				"children": [
					{
						"key": "sideBar",
						"visible": true
					}
				],
				"visible": true
			},
			{
				"key": "tempContainer-Ta1fthRDypt7Enxic3vwca",
				"children": [
					{
						"key": "noteList",
						"visible": true
					}
				],
				"visible": true
			},
			{
				"key": "editor",
				"visible": true
			}
		],
		"visible": true
	},
```
</details>

---

- Putting the plugins folder back causes this error, but I am still able to rearrange plugins without a crash. Even after deleting and reinstalling the Backlinks plugin from the UI, I still see this error, but everything seems to be working ok. 

<details>
<summary>Crash log - n reducer: Cannot read property 'key' of null Action: {"type":"MAIN_LAYOUT_SET_ITEM_PROP","itemKey":"plugin-view-joplin.plugin.ambrt.backlinksToNote-backlinksPanel","propName":"visible","propValue":true} </summary>

```
/Applications/Joplin.app/Contents/Resources/app.asar/app.js:212 Uncaught (in promise) TypeError: In reducer: Cannot read property 'key' of null Action: {"type":"MAIN_LAYOUT_SET_ITEM_PROP","itemKey":"plugin-view-joplin.plugin.ambrt.backlinksToNote-backlinksPanel","propName":"visible","propValue":true}
    at /Applications/Joplin.app/Contents/Resources/app.asar/app.js:212
    at Object.iterateItems [as default] (/Applications/Joplin.app/Contents/Resources/app.asar/gui/ResizableLayout/utils/iterateItems.js:6)
    at /Applications/Joplin.app/Contents/Resources/app.asar/app.js:211
    at Immer.produce (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/immer/dist/immer.cjs.development.js:748)
    at reducer (/Applications/Joplin.app/Contents/Resources/app.asar/app.js:210)
    at dispatch (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/node_modules/redux/lib/createStore.js:178)
    at Application.<anonymous> (/Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/BaseApplication.js:453)
    at Generator.next (<anonymous>)
    at /Applications/Joplin.app/Contents/Resources/app.asar/node_modules/@joplin/lib/BaseApplication.js:8
    at new Promise (<anonymous>)

```
</details>
